travis_ci.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+#############################################################################
+# This script is what will run in CI (Continuous Integration) to make sure  #
+# the code base is building, tests are passing, and formatting is correct   #
+#                                                                           #
+# It relies on certain bash variables being set by CI before this script is #
+# called. This setup is dictated in the `.travis.yml` file                  # 
+#############################################################################
+
+# The current directory
+CURR_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+# This variable is used to let us show nice folds in travis
+export TRAVIS_FOLD_COUNTER=1
+
+# Display command in Travis console and fold output in dropdown section
+function travis_run() {
+  local command=$@
+
+  echo -e "\e[0Ktravis_fold:start:command$TRAVIS_FOLD_COUNTER \e[34m$ $command\e[0m"
+  # actually run command
+  $command || exit 1 # kill build if error
+  echo -e -n "\e[0Ktravis_fold:end:command$TRAVIS_FOLD_COUNTER\e[0m"
+
+  let "TRAVIS_FOLD_COUNTER += 1"
+}
+
+
+# Change to the directory this script is in
+cd $CURR_DIR
+
+# Note that we must build the codebase in order to run tests
+if [ "$RUN_BUILD" == "true" ] || [ "$RUN_TESTS" == "true" ]; then
+    # Install all required dependecies
+    travis_run ./setup_scripts/install_dependencies.sh
```

Comment:
~~I just checked the install_dependencies.sh, rosinstall command is missing, iirc rosinstall should also be run in this script~~

Reply:
NVM, we can do this later, I can add it in through the other PR once we merge this one. Makes more sense that way. However it makes my previous comment about udev ruels awkward because where do we put phidget udev mods...

Reply:
This should still work I think. `setup_udev_rules.sh` is called from `install_tools.sh`, *not* `install_dependencies.sh`, so CI never calls it. I'd advocate for having the phidget rules go in `setup_scripts/10-snowbots.rules`